receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -625,3 +706,124 @@ func (prw *prometheusRemoteWriteReceiver) extractScopeInfo(ls labels.Labels) (st
+
+	if value.IsStaleNaN(histogram.Sum) {
+		dp.SetFlags(pmetric.DefaultDataPointFlags.WithNoRecordedValue(true))
+	} else {
+		setCountAndSum(histogram, dp)
+	}
+
+	dp.ExplicitBounds().FromRaw(histogram.CustomValues)
+	bucketCounts := convertCBNHBuckets(histogram)
+	dp.BucketCounts().FromRaw(bucketCounts)
+
+	extractAttributes(ls).CopyTo(dp.Attributes())
+	stats.Histograms++
+}
+
+// convertCBNHBuckets converts CBNH bucket data to OpenTelemetry bucket counts
+func convertCBNHBuckets(histogram writev2.Histogram) []uint64 {
+	// For CBNH, we need numExplicitBounds + 1 buckets (including the final +inf bucket)
+	bucketCounts := make([]uint64, len(histogram.CustomValues)+1)
+
+	// CBNH uses the positive bucket list and spans for all buckets
+	if len(histogram.PositiveSpans) == 0 {
+		return bucketCounts
+	}
+
+	if histogram.IsFloatHistogram() {
+		// Float histograms: values are absolute counts
+		bucketIdx := 0
+		for _, span := range histogram.PositiveSpans {
+			// Skip empty buckets based on offset
+			bucketIdx += int(span.Offset)
+
+			// Fill buckets for this span
+			for i := uint32(0); i < span.Length && bucketIdx < len(bucketCounts) && i < uint32(len(histogram.PositiveCounts)); i++ {
+				if bucketIdx >= 0 && bucketIdx < len(bucketCounts) {
+					bucketCounts[bucketIdx] = uint64(histogram.PositiveCounts[i])
+				}
+				bucketIdx++
+			}
+		}
+
+		// Convert from cumulative to delta buckets (like classic histograms)
+		for i := len(bucketCounts) - 1; i > 0; i-- {
+			bucketCounts[i] -= bucketCounts[i-1]
+		}
+	} else {
+		// Integer histograms: values are deltas between buckets
+		bucketIdx := 0
+		bucketCount := int64(0)
+		deltaIdx := 0
+
+		for _, span := range histogram.PositiveSpans {
+			// Skip empty buckets based on offset
+			bucketIdx += int(span.Offset)
+
+			// Fill buckets for this span
+			for i := uint32(0); i < span.Length && bucketIdx < len(bucketCounts) && deltaIdx < len(histogram.PositiveDeltas); i++ {
+				bucketCount += histogram.PositiveDeltas[deltaIdx]
+				deltaIdx++
+
+				if bucketIdx >= 0 && bucketIdx < len(bucketCounts) {
+					bucketCounts[bucketIdx] = uint64(bucketCount)
+				}
+				bucketIdx++
+			}
+		}
+
+		// Convert from cumulative to delta buckets (like classic histograms)
+		for i := len(bucketCounts) - 1; i > 0; i-- {
+			bucketCounts[i] -= bucketCounts[i-1]
+		}
+	}
+
+	// Ensure the final (+inf) bucket count so that sum of bucket counts equals overall count
+	totalBucketCount := uint64(0)
+	for i := 0; i < len(bucketCounts)-1; i++ {
+		totalBucketCount += bucketCounts[i]
+	}
+
+	var overallCount uint64
+	if histogram.IsFloatHistogram() {
+		overallCount = uint64(histogram.GetCountFloat())
+	} else {
+		overallCount = histogram.GetCountInt()
+	}
+
+	// Set the final bucket count to make the total equal to overall count
+	if overallCount >= totalBucketCount {
+		bucketCounts[len(bucketCounts)-1] = overallCount - totalBucketCount
+	}
+
```

Comment:
In general it should be the sender to ensure that the bucket count is consistent with the sum of buckets.
So I wouldn't do this. 

You can do validation that it's true and reject the sample, but I don't think we should make up numbers.

```suggestion
```